### PR TITLE
nanodbc: multi-line error message

### PR DESCRIPTION
**Pull request by detule**
Fixes https://github.com/r-dbi/odbc/issues/564

Error would now read:

```
> con <- DBI::dbConnect(odbc::odbc(), "postgresql_testing")
  Error: nanodbc/nanodbc.cpp:1135: 00000
  [unixODBC][Driver Manager]Data source name not found and no default driver specified
```

AFAICT there is no good way to break up the bottom message, as it comes through from the driver/manager in one chunk.  We could try parsing it, or making assumptions about its composition, but at that point we might be crossing the pain/gain threshold.

### CODE REVIEW

**Comment by hadley** [comment]
Is it worth adding a snapshot test for this? Or too hard to do in a way that doesn't end up mostly capturing an error generated by someone else?

**Comment by hadley** [comment]
Docs for the message specification: https://learn.microsoft.com/en-us/sql/odbc/reference/develop-app/diagnostic-messages?view=sql-server-ver16.

Agreed that it's probably not worth parsing unless we had a burning desire to format it some other way (maybe with a different colour).

File: src/nanodbc/nanodbc.cpp [hunk 1]
```diff
@@ -411,7 +411,7 @@ recent_error(SQLHANDLE handle, SQLSMALLINT handle_type, long& native, std::strin
     state = std::string(&sql_state[0], &sql_state[arrlen(sql_state) - 1]);
     native = native_error;
     std::string status = state;
-    status += ": ";
+    status += "\n";
```
[the remark this answers is not in the document]
Comment (detule) [reply src/nanodbc/nanodbc.cpp:414]: Thanks!

Turned the space @ 403 into a newline character as well.

**Comment by detule** [comment]
> Is it worth adding a snapshot test for this? Or too hard to do in a way that doesn't end up mostly capturing an error generated by someone else?

I think you are spot on with testing an error message.  I tried to add something more generic that makes sure there is at least one newline character, but perhaps I over-simplified.

**Review by hadley — APPROVED** [review]

**Comment by hadley** [comment]
Probably worth a news bullet too.

**Pull request merged**